distributed/scheduler.py

```diff
@@ -7347,6 +7494,46 @@ def get_nbytes(self, keys=None, summary=True):
 
             return result
 
+    async def get_cluster_story(self, keys=None, on_error="raise", stimulus_id=None):
+        assert isinstance(keys, tuple)
+        bits = [
+            (ws, await self.rpc.connect(ws.address)) for ws in self.workers.values()
+        ]
+        tasks = []
+
+        for _, worker_comm in bits:
+            coro = send_recv(comm=worker_comm, reply=True, op="get_story", keys=keys)
+            tasks.append(asyncio.ensure_future(coro))
+
+        try:
+            worker_stories = await asyncio.gather(
+                *tasks, return_exceptions=on_error == "ignore"
+            )
+        except Exception:
+            for task in tasks:
+                task.cancel()
+
+            raise
+        finally:
+            for worker_state, worker_comm in bits:
+                self.rpc.reuse(worker_state.address, worker_comm)
```

Comment:
I think reproducing the pattern from https://github.com/dask/distributed/blob/5c7d5550aa535ad525a842ce7796badec7ff9453/distributed/scheduler.py#L6155-L6187 would be cleaner